engine/src/main/java/com/datatorrent/stram/engine/Stream.java

```diff
@@ -32,7 +33,7 @@
 /*
  * Provides basic interface for a stream object. Stram, StramChild work via this interface
  */
-public interface Stream extends Component<StreamContext>, ActivationListener<StreamContext>, Sink<Object>
+public interface Stream extends Component<StreamContext>, ActivationListener<StreamContext>, ControlTupleEnabledSink<Object>
```

Comment:
The control tuple awareness here is for them to be able to wrap the user's control tuple in a new CustomControlTuple which is only accessible to the engine. The de-multiplexing still happens at the nodes.